### ISSUE TITLE
Python packages forced downgrade??  

### ISSUE THREAD

**Issue by MattBC90** [Esri/arcgis-python-api#999]
Hey folks, so I have an issue.  I'm new here and not an expert, so please be gentle.  

I am trying to do a bunch of deep learning stuff using ArcGIS Pro 2.7.2 and/or Jupyter Notebook importing ArcGIS libraries.  I keep getting snagged and it seems to always be related to having python packages installed into my (cloned) environment that I cannot easily update. 

The first updating issue is that I cannot update any packages inside of the options menu/python manager inside of ArcGIS Pro at all.  I click "update" and nothing happens.  Yes, this is on the cloned environment which _should_ allow for updating packages, but it doesn't.  Some of the packages include some of the DL stuff, but also default packages as well.  In this state, I cannot effectively use notebook inside of ArcGIS Pro because features like displaying batch samples simply does nothing.

So I tried to fix this by updating all of the packages from the Python Command Prompt, which unfortunately forces me to accept package downgrades.  I accepted this once, and it totally bricked notebook in AGP for me.  This is the downgrade part after the update section:

"The following packages will be DOWNGRADED:

    jupyterlab_server:   2.4.0-pyhd3eb1b0_0       --> 1.2.0-py_0
    matplotlib-base:     3.3.1-py37hba9282a_0     --> 3.2.2-py37h64f37c6_0
    vc:                  14.2-h21ff451_1          --> 14.1-h0510ff6_4

Proceed ([y]/n)?"

Since this happened, I have been more successful in using ArcGIS packages linked to Jupyter notebook in my browser, however, I have hit a new issue in my workflow that is more than likely related to an out-of-date python package.  So in other words, I have to figure out how to properly update these packages.  I don't understand what the downgrade is about but it 100% is bad for notebook.  

Any ideas? 
![AGPpythonUpdates](https://user-images.githubusercontent.com/83701093/117126507-b247be80-ad4f-11eb-9d33-75094052eb37.png)

**Comment by MattBC90** [comment]
Just looking into it, I think I have a better idea of the issue.  

Basically, I think that there are some compatibility problems going on that involve some Deep Learning conda packages that only work with py3.7, but when updating all via ArcGIS Pro python manager, it updates python to 3.9 along with only a few other packages that are compatible with 3.9 only.  Or in other words, trying to easily update packages to get them to work at all also messes up the whole environment.  

I really need to fix this and all I can think of is trying to find a compatibility table somewhere with all 49 or so packages that need updating, and seeing which specific patch fits with 3.7.10 or whatever is the highest level for deep learning at the moment (I don't think it is rated for 3.9 right now).  I can manually install individual patches via the Anaconda manager outside of ArcGIS Pro, but it only tells me which patches I can update to, not which version of python they are compatible with.  As you can imagine, this seems pretty convoluted and probably will lead to additional problems.

**Comment by Detteor** [comment]
I wrote a couple of comments about this in another thread on here.
[First Comment](https://github.com/Esri/arcgis-python-api/issues/947#issuecomment-812962566)
[Second Comment](https://github.com/Esri/arcgis-python-api/issues/947#issuecomment-813781305)


TL;DR Create a new environment with `conda create -n myenv python=3.7`. This will make sure that the packages don't upgrade the version of python. Then `conda install -c esri arcpy` or `conda install -c esri arcgis` if you don't want or have access to arcpy. After that you can set the .condarc to pull from that channel `conda config --env --add channels esri`. You can also edit/create .condarc in the Anaconda Installation folder and add the channels to that manually.

You can also try to use `conda config --env --set channel_priority strict` **BE CAREFUL** to install the esri packages before you do this since the metadata for the esri packages likes to grab stuff from a lot of different channels. I would do this if for some reason you want to have geopandas or other Open Source GIS packages in the same environment. Another caveat is that ESRI does have some versions of the Open Source GIS Packages in their channel, which can be very outdated.

[Condarc Docs on Anaconda Website](https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html)

**Comment by MattBC90** [comment]
I eventually found some info last night about how the ESRI Deep Learning packages don't support RTX 3000 series GPUs yet, which I am using, and they listed many of the technical problems I was encountering.  So it appears to be a software-hardware compatibility issue, which is incredibly lame to be honest since these cards have been out for the better part of a year.

The other issue is that the ESRI Deep Learning framework packages are not part of the included ESRI conda/python environment- they have to be added in.  The programming behind Deep Learning is largely open-source, but what ESRI appears to be doing is putting a lot of the more advanced stuff under the hood in the arcgis library.  

So if I was going to go the route of just using Anaconda, I would just use the open source stuff that should work better without ESRI libraries at all.  Basically this is all to derive a trained model using configuration not available in ArcGIS Pro toolset, but the resulting .dlpk file should be universal and usable inside of ArcGIS Pro regardless of where it came from.  

I will take note of those commands though, as they appear to be generally useful.  I'm starting to see why the older GIS people are always paranoid about version updates heh.

**Comment by Detteor** [comment]
> I eventually found some info last night about how the ESRI Deep Learning packages don't support RTX 3000 series GPUs yet, which I am using, and they listed many of the technical problems I was encountering. So it appears to be a software-hardware compatibility issue, which is incredibly lame to be honest since these cards have been out for the better part of a year.

Good to know. I personally use a 2080 for most of my Deep Learning needs.

When I first started to use the DL capabilities in 2.6, I started by using the [Deep Learning Framework](https://github.com/Esri/deep-learning-frameworks) from ESRI. After becoming very frustrated by it and also wanting to use Linux for certain task, I noticed that they started to release metapackages on anaconda. These packages help to keep your environment stable and make sure that is continues to function with the most recent version of ArcGIS Pro.

[MetaPackage for Windows](https://anaconda.org/Esri/deep-learning-essentials/files)
[MetaPackage for Linux](https://anaconda.org/Esri/arcgis_learn/files)

You can download the bz2 files and open with 7Zip to see what the index.json has on it to see what they are pinning the packages to.

> EDIT: for the "channel_priority strict" line, are you saying install arcgis, arcpy and the Deep Learning frameworks first and then use this line to install the rest of the ESRI package suite?

Yup I have found that the arcpy install will freeze if using strict since not all the packages are provided by the esri channel. When you install arcpy it will install the latest version of  arcgis as well since it is pinned at 1.8.* in the metadata.

**Comment by priyankatuteja** [comment]
@MattBC90 My understanding is that you are trying upgrade arcgis version in your cloned environment of pro 2.7.2. The desired arcgis version is 1.8.5 and when you go ahead and update it, some packages get downgraded and you are not able to run Pro Embedded Notebook successfully, w.r.t displaying images using show_batch, lr_find or show_results. Is my understanding correct?

**Comment by AtmaMani** [comment]
Closing due to inactivity
